### PR TITLE
Test coverage providers

### PR DESCRIPTION
**Pull request by jmauerhan**
- Added tests for all Providers. All tests are isolated correctness/contract/collaboration tests. 
- Moved the AspectMock teardown into the TestCase teardown. This won't affect anything that isn't using AspectMock, and will prevent forgetting to clean AspectMock when a test is over.

### CODE REVIEW

**Comment by jmauerhan** [comment]
As a note, some of these things were covered by other tests, but those tests are not isolated - which means there were not any specific tests for these item's contracts and collaboration with their dependencies. I find that these specific isolated tests provide better security when refactoring and make it easier to make non-breaking changes to both the consumers and providers for the system they test. So this PR does not add a ton of new coverage, but was intended to add improved quality of the coverage. 

Thanks :)

File: tests/src/Provider/Node/SystemNodeProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testGetNodeReturnsSystemNodeFromMacAddress()
+    {
+        $provider = new SystemNodeProvider();
+        $node = $provider->getNode();
+
+        $this->assertTrue(ctype_xdigit($node), 'Node is not a hexadecimal string. Actual node: ' . $node);
+        $this->assertTrue(strlen($node) === 12, 'Node is 12 characters long. Actual length: ' . strlen(
```
Comment (ramsey) [inline tests/src/Provider/Node/SystemNodeProviderTest.php]: Should the message be "Node is *not* 12 characters long." ?

Comment (aztech-dev) [inline tests/src/Provider/Node/SystemNodeProviderTest.php]: or "Node *should* be 12 characters long, got %d instead"

**Comment by ramsey** [comment]
@jmauerhan, I've reviewed the pull request, and all looks great. I think the only change is that test message mentioned above. Once you make that change, I'll merge it in. Thanks!

File: tests/src/Provider/Node/SystemNodeProviderTest.php (hunk 1 above)
Comment (jmauerhan) [inline tests/src/Provider/Node/SystemNodeProviderTest.php]: Good catch guys thanks 😊 I'll make that asap

**Pull request merged**